vignettes/simulations.Rmd

```diff
@@ -0,0 +1,851 @@
+
+Since this is simple randomization, it does not take into account the initial covariates, and treatment assignment occurs randomly.
+
+```{r, simple-result}
+# simple randomization
+simple_results <-
+  function(data, arms, ratio) {
+    for (i in unique(data$simnr))
+    {
+      for (j in 1:n)
+      {
+        current_data <- data[data$simnr == i, ]
+        
+        start <- 1 + (i - 1) * n
+        end <- i * n
+        
+        data[start:end,]$arm[j] <-
+           randomize_simple(arms, ratio)
+        
+        
+      }
+    }
+    return(data)
+  }
+```
+
+The table illustrates an example of data output for simple randomization in the first iteration.
+
+```{r, simple-data}
+simple_data <-
+  simple_results(data, c("armA", "armB", "armC"), c("armB" = 1L,"armA" = 1L, "armC" = 1L))
+
+simple_data[1:5, 2:ncol(simple_data)] |>
+  gt()
+```
+The table presents a statistical summary of results for the first iteration.
+
+```{r, chi2-4, tab.cap = "Summary of proportion test for simple randomization"}
+statistics_table(simple_data)
+```
+
+## Block randomization
+
+Block randomization, as opposed to minimization and simple randomization methods, was developed based on the `rbprPar` function available in the `randomizeR` package. Using this, the `block_rand` function was created, which, based on the defined number of patients, arms, and a list of stratifying factors, generates a randomization list with a length equal to the number of patients multiplied by the product of categories in each covariate. In the case of the specified data in the document, for one iteration, it amounts to **105 * 2^6 = 6720 rows**.
+
+```{r, block-rand}
+# Function to generate a randomisation list
+block_rand <-
+  function(N, block, n_groups, strata) {
+    strata_grid = expand.grid(strata)
+    
+    strata_n = nrow(strata_grid)
+    
+    ratio = rep(1, n_groups)
+    
+    arm_names = LETTERS[1:n_groups]
+    
+    genSeq_list <- lapply(seq_len(strata_n), function(i) {
+      rand <- rpbrPar(
+        N = N,
+        rb = block,
+        K = n_groups,
+        ratio = ratio,
+        groups = arm_names,
+        filledBlock = FALSE
+      )
+      getRandList(genSeq(rand))[1,]
+    })
+    df_list = tibble::tibble()
+    for (i in seq_len(strata_n)) {
+      local_df <- strata_grid %>%
+        dplyr::slice(i) %>%
+        dplyr::mutate(count = N) %>%
+        tidyr::uncount(count) %>%
+        tibble::add_column(rand_arm = genSeq_list[[i]])
+      df_list <- rbind(local_df, df_list)
+    }
+    return(df_list)
+  }
+```
+
+Lists of randomization codes were generated for each iteration based on the function, assuming a block vector of c(3,6,9). This ensures blinding and facilitates ease in predicting which group the next patient would be assigned to.
+
+```{r, rand-data}
+# generate randomization lists for each iterations
+rand_data <- tibble::tibble()
+
+for (i in unique(data$simnr)) {
+  simulation_result <-   
```

Comment:
you can make this a function instead and generate randomization table on demand to conserve memory